src/lib/db/group-store.ts

```diff
@@ -190,25 +209,31 @@ export default class GroupStore implements IGroupStore {
 
     async addUsersToGroup(
         groupId: number,
-        users: IGroupUserModel[],
+        users: ICreateGroupUserModel[],
         userName: string,
-        transaction?: Transaction,
     ): Promise<void> {
-        const rows = (users || []).map((user) => {
-            return {
-                group_id: groupId,
-                user_id: user.user.id,
-                created_by: userName,
-            };
-        });
-        return (transaction || this.db).batchInsert(T.GROUP_USER, rows);
+        try {
+            const rows = (users || []).map((user) => {
+                return {
+                    group_id: groupId,
+                    user_id: user.user.id,
+                    created_by: userName,
+                };
+            });
+            return await this.db.batchInsert(T.GROUP_USER, rows);
+        } catch (error) {
+            if (
```

Comment:
<img width="433" alt="Screenshot 2023-07-24 at 09 22 02" src="https://github.com/Unleash/unleash/assets/1394682/1c4bc9fd-4692-4ef3-8ebd-82410f301df2">
If user list has invalid user id. We could check this upfront when we validate group in the service but it was more invasive since we'd need to inject read models from the user subdomain. I don't mind having this code here since it's DB related code wrapped in a DB specific store so the details are not leaking out.

Reply:
Yeah agreed, I think it's fine living in this logical layer